### PR TITLE
feat: Return Profile images for members from resources

### LINKED ISSUE #411: feat: Include mentor profile picture when returning all mentors
## 📸 User Story

**As a** platform user  
**I want to** see mentor profile pictures in the mentor list  
**So that** I can visually identify mentors and have a more engaging browsing experience

---

### AC1: Retrieve member profile picture from ResourceService

- **Given** a mentor has a profile picture stored in the resources system  
- **When** the `getAllMentors` endpoint is called  
- **Then** the response should include the mentor's profile picture URL from the ResourceService  
- **And** the profile picture should be retrieved using the existing `member-profile-picture` resource type
- **And** in the response the attribute images should be return like this example: 
```
"images": [
      {
        "path": "driveFileLink_from_resource",
        "alt": "mentor_name profile photo",
        "type": "DESKTOP"
      }
    ]
```

### AC2: Handle mentors without profile pictures

- **Given** a mentor does not have a profile picture in the resources system  
- **When** the `getAllMentors` endpoint is called  
- **Then** the response should include the mentor without profile picture  
- **And** the API should not fail or throw errors

### AC3: Reuse existing ResourceService methods

- **Given** `ResourceService` already has methods to retrieve member profile pictures  
- **When** implementing this feature in `MentorshipService`  
- **Then** existing `ResourceService` methods must be reused without duplication  
- **And** no new methods should be created for retrieving profile pictures

---

## 🔧 Technical Notes

- Reuse existing `ResourceService` methods for retrieving `MEMBER_PROFILE_PICTURE` resources
- Ensure proper null handling when profile pictures don't exist
- Update response DTO to include profile resource 

---

## 🧪 Testing Suggestions

- [ ] Unit tests for `MentorshipService` updated
- [ ] Integration tests updated
- [ ] Test edge case: mentor without profile picture
- [ ] Verify OpenAPI/Swagger documentation is updated

---

## 🔗 Related

- API Mentors: [`GET mentors`](https://wcc-backend.fly.dev/swagger-ui/index.html#/Platform/getAllMentors)
- API Resources: [`GET getMentorProfilePicture`](https://wcc-backend.fly.dev/swagger-ui/index.html#/Resources/getMentorProfilePicture)

### PR DESCRIPTION
**Pull request by dricazenck**
## Description

1. feat: Populate images field with profile picture from member_profile_picture
2. refactor: Remove member_images table and related code

Details:
- Inject MemberProfilePictureRepository into MemberService and MentorshipService
- Add helper methods to convert Resource to Image
- Update getAllMembers() to enrich members with profile pictures
- Update getAllMentors() to enrich mentors with profile pictures
- Add unit tests for profile picture merging logic
- Add integration tests for end-to-end verification

## Related Issue

Closes #411 

## Change Type

- [x] New Feature
- [x] Code Refactor

## Screenshots

image

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] I checked and followed the [contributor guide](../CONTRIBUTING.md)
- [x] I have tested my changes locally.

### CODE REVIEW

**Review by nverbic — APPROVED** [review]
Approved!

File: src/main/java/com/wcc/platform/service/MemberService.java [hunk 1]
```diff
@@ -41,7 +49,7 @@ public List<Member> getAllMembers() {
     if (allMembers == null) {
       return List.of();
     }
-    return allMembers;
+    return allMembers.stream().map(this::enrichWithProfilePicture).toList();
```
[the remark this answers is not in the document]
Comment (dricazenck) [reply src/main/java/com/wcc/platform/service/MemberService.java]: All members can have profile, but probably only mentors are going to have in the beginning. But the code will be ready for that.
